api/envoy/extensions/filters/network/thrift_proxy/v3/thrift_proxy.proto

```diff
@@ -76,7 +88,12 @@ message ThriftProxy {
   string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
 
   // The route table for the connection manager is static and is specified in this property.
-  RouteConfiguration route_config = 4;
+  // It is invalid to define both route_config and trds.
+  RouteConfiguration route_config = 4
+      [(udpa.annotations.field_migrate).oneof_promotion = "route_specifier"];
+
+  // Use xDS to fetch the route configuration. It is invalid to define both route_config and trds.
+  Trds trds = 8 [(udpa.annotations.field_migrate).oneof_promotion = "route_specifier"];
```

Comment:
Please add an entry to version history for this new feature.

Reply:
Done